core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2980,6 +2980,77 @@ private[spark] object Utils extends Logging {
     metadata.toString
   }
 
+  /**
+   * Download Ivy URIs dependent jars.
+   *
+   * @param uri Ivy uri need to be downloaded.
+   * @return Comma separated string list of URIs of downloaded jars
```

Comment:
Should we be returning a `List[String]` instead of `String` (here and in `SparkSubmitUtils`)? It seems odd to have `SparkSubmitUtils` do a `mkString` to convert a list to string, then re-convert back to a list later.

Reply:
> Should we be returning a `List[String]` instead of `String` (here and in `SparkSubmitUtils`)? It seems odd to have `SparkSubmitUtils` do a `mkString` to convert a list to string, then re-convert back to a list later.

Here return `String` since `SparkSubmit.resolveMavenCoordinates` return `String` if we change these method will change a lot and it's not related to current. pr, how about start a new pr after or before current pr to refactor these code 
 about return `Array[String]`?

Reply:
Sure, makes sense.